homeassistant/components/n26/sensor.py

```diff
@@ -0,0 +1,270 @@
+        self._client = client
+        self._account_info = account_info
+        self._account_statuses = account_statuses
+        self._limits = limits
+        self._balance = balance
+        self._iban = balance["iban"]
+
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        self._account_info = self._client.get_account_info()
+        self._account_statuses = self._client.get_account_statuses()
+        self._limits = self._client.get_account_limits()
+        self._balance = self._client.get_balance()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the entity."""
+        return self._iban[-4:]
+
+    @property
+    def name(self) -> str:
+        """Friendly name of the sensor."""
+        return "n26_{}".format(self._iban[-4:])
+
+    @property
+    def state(self) -> float:
+        """Return the balance of the account as state."""
+        if self._balance is None:
+            return None
+
+        return self._balance.get("availableBalance")
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Use the currency as unit of measurement."""
+        if self._balance is None:
+            return None
+
+        return self._balance.get("currency")
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Additional attributes of the sensor."""
+        attributes = {
+            ATTR_IBAN: self._balance.get("iban"),
+            ATTR_BANK_BALANCE: self._balance.get("bankBalance"),
+            ATTR_USABLE_BALANCE: self._balance.get("usableBalance"),
+            ATTR_ACC_OWNER_TITLE: self._account_info.get("title"),
+            ATTR_ACC_OWNER_FIRST_NAME: self._account_info.get("kycFirstName"),
+            ATTR_ACC_OWNER_LAST_NAME: self._account_info.get("kycLastName"),
+            ATTR_ACC_OWNER_GENDER: self._account_info.get("gender"),
+            ATTR_ACC_OWNER_BIRTH_DATE: timestamp_ms_to_date(
+                self._account_info.get("birthDate")),
+            ATTR_ACC_OWNER_EMAIL: self._account_info.get("email"),
+            ATTR_ACC_OWNER_PHONE_NUMBER:
+                self._account_info.get("mobilePhoneNumber"),
+        }
+
+        for limit in self._limits:
+            limit_attr_name = "limit_{}".format(limit["limit"].lower())
+            attributes[limit_attr_name] = limit["amount"]
+
+        return attributes
+
+    @property
+    def icon(self) -> str:
+        """Set the icon for the sensor."""
+        return ICON_ACCOUNT
+
+
+class N26Card(Entity):
+    """Sensor for a N26 card."""
+
+    def __init__(self, client, account_name, card) -> None:
+        """Initialize a N26 card."""
+        self._client = client
+        self._account_name = account_name
+        self._card = card
+
+    @Throttle(min_time=SCAN_INTERVAL * 0.8)
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        self.hass.data[DOMAIN][CARDS] = self._client.get_cards()
```

Comment:
It would be better if we centralized this updating to another single instance. And then we fetch from that cache from all entities. Now each entity will call the client which isn't necessary.

So create a single data instance with an update method and put the throttle on there instead. Look at other components for examples.

Reply:
I have to say it's a little bit vague imo to look at "other components" for this since there are a **lot** of them and many do not adhere to the current coding guidelines. A more specific hint wouldn't hurt :smile: 

I have reworked the code similar to how it was done in the yahoo weather component. Please have a look and let me know if anything is wrong.